Source/ForecastDeskbarView.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2017 Benjamin Amos
+ * All rights reserved. Distributed under the terms of the MIT license.
+ */
+
+#include <Bitmap.h>
+#include <StringView.h>
+#include <String.h>
+#include <MessageRunner.h>
+#include <Looper.h>
+#include <Roster.h>
+
+#include "ForecastDeskbarView.h"
+#include "ForecastView.h"
+#include "App.h"
+#include "Util.h"
+
+const uint32 kUpdateForecastMessage = 'Updt';
+const float kToolTipDelay = 1000000; /*1000000ms = 1s*/
+
+ForecastDeskbarView::ForecastDeskbarView(BRect viewSize, ForecastView* forecastView)
+	:	BView(viewSize, "ForecastDeskbarView", B_FOLLOW_ALL, B_WILL_DRAW | B_NAVIGABLE | B_PULSE_NEEDED)
+{
+	fForecastView = forecastView;
+}
+
+ForecastDeskbarView::ForecastDeskbarView(BMessage* archive)
+	:	BView(BRect(0, 0, 15, 15), "ForecastDeskbarView", B_FOLLOW_ALL, B_WILL_DRAW)
+{
+	fForecastView = NULL;
+}
+
+ForecastDeskbarView::~ForecastDeskbarView()
+{
+	delete fMessageRunner;
+}
+
+void
+ForecastDeskbarView::AttachedToWindow()
+{
+	fMessageRunner = new BMessageRunner(BMessenger(this), new BMessage(kUpdateForecastMessage), kToolTipDelay, -1);
+	fForecastView->AttachedToWindow();
```

Comment:
This is not needed, AddChild(fForecastView) will do it.
It is also a bit strange to have an fForecastView here, if you are not actually drawing anything. We should probably extract some code from fForecastView into an utility class.

Reply:
In this method you should use AdoptParentColors() so you remove the white background from the replicant.